be/src/exec/csv_scanner.cpp

```diff
@@ -506,4 +514,45 @@ void CSVScanner::_report_rejected_record(const CSVReader::Record& record, const
     _state->append_rejected_record_to_file(record.to_string(), err_msg, _curr_reader->filename());
 }
 
+static TypeDescriptor get_type_desc(const Slice& field) {
+    StringParser::ParseResult result;
+
+    StringParser::string_to_int<int64_t>(field.get_data(), field.get_size(), &result);
+    if (result == StringParser::PARSE_SUCCESS) {
+        return TypeDescriptor(TYPE_BIGINT);
+    }
+
+    StringParser::string_to_float<double>(field.get_data(), field.get_size(), &result);
+    if (result == StringParser::PARSE_SUCCESS) {
+        return TypeDescriptor(TYPE_DOUBLE);
+    }
+
+    StringParser::string_to_bool(field.get_data(), field.get_size(), &result);
+    if (result == StringParser::PARSE_SUCCESS) {
+        return TypeDescriptor(TYPE_BOOLEAN);
+    }
+
+    // default VARCHAR.
+    return TypeDescriptor::create_varchar_type(TypeDescriptor::MAX_VARCHAR_LENGTH);
+}
+
+Status CSVScanner::get_schema(std::vector<SlotDescriptor>* schema) {
```

Comment:
I'm curiouse why we like using pointer instead of reference?

Reply:
It is the recommended code style.